src/api/goods.ts

```
@@ -8,9 +8,11 @@ export function getAll(): Promise<Good[]> {
 }
 
 export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+  return getAll()
+    .then(goods => [...goods].sort((a, b) => a.name.localeCompare(b.name)))
+    .then(goods => goods.slice(0, 5));
 };
 
 export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+  return getAll().then(goods => goods.filter(el => el.color === 'red'));
```

This also violates checklist item #3: "[API] - When you make a request to the server, don't forget to handle possible errors". `getRedGoods` depends on `getAll` but does not handle failures itself; it expects `getAll` to reject on error. Make `getAll` throw on non-OK responses (as above) so `getRedGoods()` will propagate a rejection that the UI can catch and display an error. Ensure the error is not swallowed here so the app can show an appropriate message. (See `getRedGoods` implementation in src/api/goods.ts.)